### PR TITLE
HOLO-969: Prevent Network Monitor from getting stuck in some blocks

### PR DESCRIPTION
**Pull request by alexanderattar**
## Describe Changes

- Added try/catch block to calls to async functions to ensure promises are resolved/rejected
- Speed up getLogs by filtering by topics
- Updated retry function so it throws an error instead of running indefinitely if max attempts is exceeded
- Added some additional logging

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] Code styles have been enforced
- [x] I have checked eslint

### CODE REVIEW

File: src/utils/network-monitor.ts [hunk 1]
```diff
@@ -979,9 +984,17 @@ export class NetworkMonitor {
     if (this.blockJobs[network].length > 0) {
       const blockJob: BlockJob = this.blockJobs[network][0] as BlockJob
       if (this.enableV2) {
-        this.processBlock2(blockJob)
+        try {
+          await this.processBlock2(blockJob)
+        } catch (error: any) {
+          this.structuredLogError(blockJob.network, `Error processing block: ${JSON.stringify(error)}`, blockJob.block)
```
Comment (alexanderattar) [inline src/utils/network-monitor.ts]: Not sure this is necessary. I will attempt to make the error logs more consistent on next pass over

**Review by ACC01ADE — APPROVED** [review]

File: src/utils/network-monitor.ts [hunk 2]
```diff
@@ -1447,19 +1474,35 @@ export class NetworkMonitor {
...
+          try {
+            await this.processTransactions2?.bind(this.parent)(job, interestingTransactions)
+          } catch (error: any) {
+            this.structuredLogError(job.network, `Error processing transactions ${error}`, job.block)
+          }
         }
 
-        this.blockJobHandler(job.network, job)
+        try {
+          return await this.blockJobHandler(job.network, job)
```
Comment (sogoiii) [inline src/utils/network-monitor.ts]: we have this method `blockJobHandler` called a few different times based on this if else tree. I wonder if we can organize it so it is only called once.

File: src/utils/network-monitor.ts [hunk 3]
```diff
@@ -2092,54 +2120,45 @@ export class NetworkMonitor {
...
+    // If canFail is true, the loop will terminate after the specified number of attempts.
+    // If canFail is false, the loop will continue until maximum attempts has been hit and then throw an error.
+    let counter = 0
 
-          return null
-        }
-      } else {
-        if (!sent) {
-          sent = true
+    while (counter < attempts) {
```
Comment (sogoiii) [inline src/utils/network-monitor.ts]: I thought we decided to use the for loop version. Is this change needed now?

**Review by sogoiii — APPROVED** [review]
👍
